collections/ansible_collections/cloudkit/config_as_code/README.md

```diff
@@ -0,0 +1,225 @@
+  lightspeed:
+    disabled: true
+EOF
+
+oc apply -f aap.yml
+```
+
+### Apply CloudKit configuration on AAP instance
+
+#### Prepare OpenShift environment
+
+Create service account and secrets:
+
+```
+cat << EOF > cloudkit_env.yml
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cloudkit-sa
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cloudkit-sa
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+  - kind: ServiceAccount
+    name: cloudkit-sa
+    namespace: fulfillment-aap
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <your AAP organization>-<your AAP project>-config-as-code-ig
+type: Opaque
+stringData:
+  RH_USERNAME: <your Red Hat username>
+  RH_PASSWORD: <your Red Hat password>
+  AAP_HOSTNAME: https://fulfillment-fulfillment-aap.apps-crc.testing
+  AAP_USERNAME: admin
+  AAP_PASSWORD: $(oc get secret/fulfillment-admin-password -n fulfillment-aap -o go-template='{{index .data "password" | base64decode}}')
+  AAP_VALIDATE_CERTS: false
+  AAP_ORGANIZATION_NAME: <the organization name to be created in AAP>
+  AAP_PROJECT_NAME: <the project name to be create in AAP>
+  AAP_PROJECT_GIT_URI: <the git repository where your playbooks and rulebooks live>
+  AAP_PROJECT_GIT_BRANCH: <the git branch to be tracked>
+  AAP_EE_IMAGE: <the execution environment image>
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <your AAP organization>-<your AAP project>-cluster-fulfillment-ig
+type: Opaque
+stringData:
+  <setup here the credentials required for cluster fulfillment playbooks (AWS, OpenStack, ...)>
+EOF
+
+oc apply -f cloudkit_env.yml -n fulfillment-aap
+```
+
+#### Option 1: from your local environment
+
+Set your environment (same as above in `<your AAP organization>-<your AAP
+project>-config-as-code-ig` secret):
+
+```
+export RH_USERNAME=...
+export RH_PASSWORD=...
+export AAP_HOSTNAME=...
+export AAP_USERNAME=...
+export AAP_PASSWORD=...
+export AAP_VALIDATE_CERTS=...
+export AAP_ORGANIZATION_NAME=...
+export AAP_PROJECT_NAME=...
+export AAP_PROJECT_GIT_URI=...
+export AAP_PROJECT_GIT_BRANCH=...
+export AAP_EE_IMAGE=...
+```
+
+Run subscription and configure playbooks:
+```
+ansible-playbook cloudkit.config_as_code.subscription
+ansible-playbook cloudkit.config_as_code.configure
+```
+
+#### Option 2: as a job running in Kubernetes
+
+If this collection is backed in a container image (e.g.: in the custom
+execution environment used in AAP), there is the possibility to run the
+configuration in a k8s job:
+
+```
+cat << EOF > job.yml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: aap-bootstrap
+spec:
+  template:
+    spec:
+      containers:
+        - image: <your container image with cloudkit.config_as_code collection in it>
+          name: boostrap
+          args:
+            - ansible-playbook
+            - "cloudkit.config_as_code.subscription"
+            - "cloudkit.config_as_code.configure"
```

Comment:
Is there a particular version of Ansible required to make this work? Trying to run these playbooks using their fully qualified names with Ansible 2.18.5 from the `cloudkit-aap` directory results in:

```
ERROR! the playbook: cloudkit.config_as_code.subscription could not be found
```

Reply:
You can try `ANSIBLE_COLLECTIONS_PATHS=collections/ ansible-playbook cloudkit.config_as_code.subscription` . The container image is configured to look for the collection at the right place.

Reply:
I added the `collection_paths` setting to the `ansible.cfg`